### ISSUE TITLE
Download option only shows 360p (Standard Definition)

### ISSUE THREAD

**Issue by LaVey-Charkus-Veros** [YePpHa/YouTubeCenter#1636]
Hi there.

Since a few days ago I'm only able to download videos at 360p (Standard Definition). I've tried switching DASH playback on and off, and changing the player type back and forth from Flash to HTML5, both to no avail. I usually set the download button to get videos at 720p, but now all options are gone except for 360p.

I'm not sure if this really classifies as an "issue" or if it's a problem on my end. I'm not very familiar with the technicalities of video files, but I thought only 480p and 1080p split the videos between two files, one for video and the other for audio.

I've been using Firefox v35 up until now, but today I updated to v37 and also updated all my extensions and scripts, and still the problem persists.

Hope this can be fixed, thanks for reading.

- Firefox v35 and v37
- YouTube Center v2.1.1

**Comment by Zarggg** [comment]
This is a known issue/limitation caused by fundamental changes in YouTube works. Since YTC's download functionality simply requests the video file from YouTube's servers and passes it to your browser's Save window, it is only able to download videos that are not in DASH format. DASH videos actually store the audio and video streams separately, then multiplex them in the player itself. In short: downloads for resolutions 480 and higher are not available because those resolution are served with DASH.

There are browser addons that are able to get around this issue by employing a local installation (on your machine) of FFmpeg, such as [CYS Audio/Video Downloader](http://www.cys-audiovideodownloader.com/). There was discussion in the past of including FFmpeg with this addon and adding code to employ a similar method of downloading the individual streams and multiplexing to a single MP4 file, but I don't know if @YePpHa is still considering or working on that.

See the following Issues for more information: #98 #174 #716 #1133

**Comment by Inyokus** [comment]
What resolutions are available for download should only be restricted by which video formats YouTube offers. Your YTC settings, browser etc. should not make a difference here.

I can not confirm this being an issue.

Are you sure, that you're issue is present for all (or most) videos on YouTube? The video formats that are available can differ depending on whether the video is by Vevo, it has been uploaded recently, it has been uploaded a long time ago, it is watched by very few people or hasn't been watched for a very long time and last but least YouTube's general mood (both company and platform).

**Comment by ErdeFB** [comment]
Besides all that has been said, please update your YTC version. 2.1.1 is ancient.

**Comment by LaVey-Charkus-Veros** [comment]
Thanks for answering, Zarggg. What puzzles me is that I've been able to download 720p videos for years using YTC without problems, and it's up until a few days ago that all higher resolution options suddenly disappeared. I also use an extension called Video DownloadHelper (VDH), but it also stopped working (I believe it happened the very same day as YTC), so I had to do both, update Firefox to v37 and update VDH. With it, now I can download both files (video and audio) of a video and it converts them into a single video file. What I didn't like is that only 1080p and higher resolutions are available (no 720p), and the converting process takes a lot of time (about 25 minutes to convert a 500Mb file).

Also, thanks for the links. I'll be sure to read them entirely. I searched first but couldn't find any issue report related to this matter, so that's why I opened this thread.

Thanks to Inyokus and ErdeFB for their replies too. By the way, in Firefox's Add-ons panel, it says YTC is v2.1.7, but if I open the options, it says it's v2.1.1. I've tried using FF's automatic updater and installing the XPI file for the latest release manually, but it stays the same (unless I'm looking in the wrong place).

Thanks for replying.

**Comment by Zarggg** [comment]
Regarding YTC version: 2.1.7 is the last version that was posted to addons.mozilla.org, which is why Firefox can't update it automatically. Until 2.1.8 is released, you will need to get updates from Github.

**Comment by LaVey-Charkus-Veros** [comment]
Thanks for the link, Zarggg.

**Comment by KamranMackey** [comment]
Closing, since looks like you guys fixed the issue yourselves.

**Comment by LaVey-Charkus-Veros** [comment]
I wouldn't say the issue was fixed, as higher definition formats are still not available to download with YTC, and there's also no workaround or converting option using this extension/script.

I just leave it for anyone's consideration if they want to re-open this thread again.

**Comment by KamranMackey** [comment]
@LaVey-Charkus-Veros Uh, screenshot of the issue?

**Comment by KamranMackey** [comment]
@LaVey-Charkus-Veros Alright, I tried to reproduce the issue, but I can't. It gives me all possible download formats, including 720p & higher.

**Comment by LaVey-Charkus-Veros** [comment]
Sure, here you go. :)

![youtube center - 01](https://cloud.githubusercontent.com/assets/8993158/7081591/288ddb3a-df0a-11e4-86d0-33773796daa6.jpg)
![youtube center - 02](https://cloud.githubusercontent.com/assets/8993158/7081592/2893110e-df0a-11e4-85cf-3e23e34c6dca.jpg)
![youtube center - 03](https://cloud.githubusercontent.com/assets/8993158/7081593/28958272-df0a-11e4-98af-50ff85913e5e.jpg)
![youtube center - 04](https://cloud.githubusercontent.com/assets/8993158/7081594/2896b070-df0a-11e4-86b1-865dd8479e2d.jpg)
![youtube center - 05](https://cloud.githubusercontent.com/assets/8993158/7081595/289b838e-df0a-11e4-9b5b-a11920ff2bb2.jpg)

**Comment by KamranMackey** [comment]
@LaVey-Charkus-Veros Hmm, that's really weird. Best to wait for @YePpHa. He will probably know how to fix your issue.

**Comment by LaVey-Charkus-Veros** [comment]
Alright. Thanks for your concern. I hope this can be fixed soon.

**Comment by Inyokus** [comment]
@KamranMackey 720p **and higher**? I'm kinda confused, since as far as I know, YouTube stopped offering videos in non-DASH for resolutions higher than 720p.

**Comment by KamranMackey** [comment]
@Inyokus Uh, I never turned DASH playback off. I still have it enabled, and I don't plan on turning it off. I used to disable it, until I found out that you can only go to 720p with DASH playback off.

**Comment by Inyokus** [comment]
@KamranMackey I'm not sure if we're talking about the same thing. At least I am referring to the resolutions that YTC offers to download at the download button under the video, not to the resolutions the player offers for playback at the gear button. Only the range of the latter is affected by enabling/disabling DASH playback.

For technical reasons, YTC can only offer resolutions that are available in non-DASH for downloading. And as far as I know, there are only 720p, 480p and possibly lower resolutions available in non-DASH by YouTube. That's why I was confused by you saying that you could *download* resolutions higher than 720p.

**Comment by Yonezpt** [comment]
You can download any video resolution as long as the video itself offers them. What you can't do is download muxed streams that aren't in 360p, 480p or 720p quality (aka non-DASH streams), all of the other qualities (and equivalent) exist individually; this means that they can only be downloaded with audio and video in seperated files.

Currently (and I hope I am not wrong about this) YTC only offers download for the muxed files and an audio file (in case people just want a podcast or song), all of the other higher qualities are better off being downloaded via 3rd party software which can join both files into one on the server side, such as http://www.clipconverter.cc/

**Comment by LaVey-Charkus-Veros** [comment]
@Yonezp Thanks for replying. Unfortunately the problem persists on my end. I've tried all of these methods:

- I reset all of YTC settings to their defaults.
- I removed YTC completely and also uninstalled Greasemonkey. Then restarted my PC and installed them again.
- I restored my system to the oldest date available on my PC (Mar. 23, 2015), which is before I started to have this issue.
- I made a backup of my Firefox profile files and completely uninstalled it; then installed it again and restarted my PC.

The only other method I can think of trying next, is to remove and reinstall Firefox again and keep it vanilla (without using my backup profile files), which is quite drastic, as some of the extensions I have are no longer available.

By the way, another issue I've been experiencing, is that the "Scroll to Player" button has disappeared, even if it's properly configured in YTC settings (issue #1615), and it seems this started to happen the same day I got only the 360p download option.

Maybe it's worth to mention that recently I had a strange problem with Firefox. One day, all of YouTube was filled with ads and opened a lot of advertisement tabs (even if using AdBlock and have my antivirus working properly). I found out that an unidentified extension had installed by itself without my permission. I removed it and the ads were gone. I suspect the 360p and scroll button gone issues may have started by that time too, although I can't be sure.

Anyway, thanks for your concern and I hope someone can find a solution.

Bye.

**Comment by LaVey-Charkus-Veros** [comment]
EUREKA, I finally got it to work!

As I mentioned before, this issue started sometime soon after I got those ads on my browser. Below one of those ads, it had a message saying "Ads by Key". So I searched that on Google and it happened to be a malware virus, so I went to some great extents trying to remove it:

1. Opened msconfig.exe and checked safe boot.
2. Restarted the PC and, now in safe mode, restored it to the oldest date available.
3. Still in safe mode, uninstalled Firefox completely using Revo Uninstaller Pro, deleted all leftovers and also deleted the [USER]/AppData/Roaming/Mozilla files manually.
4. Restarted my PC in normal mode and ran three malware removing programs, following the steps provided in this link: http://malwaretips.com/blogs/ads-by-key-coupons-removal/
5. Restarted my PC again and installed a fresh version of Firefox and the most recent versions of both Greasemonkey and YouTube Center. Then checked YouTube and the option to download 720p was there!
6. Now I'm currently going through the hassle of installing and configuring all of my extensions, scripts and themes one by one, and also configuring Firefox the way I like it to work. Fortunately, I made lots of backups and took many screenshots to help me out.

Now I finally get the option to download in 720p or higher, and I can also watch videos in 720p with DASH turned off!

What puzzles me is that all three malware removing programs found quite a lot of malware in my PC, as I'm VERY CAREFUL about which software I install and always take good care of doing customized installations and opt out of any additional suggested installs.

It seems that something got screwed inside my profile folder, as in one of my testings, after I already had the 720p option back, I restored my original profile folder and got the 360p issue again. So, to sum it up, it seems all I had to do was to get rid of the old profile folder and let Firefox create a new one. And, in retrospective, I can say that YouTube Center had nothing to do with the problem, other than being affected by it.

The "Scroll to Player" button is still missing, though (issue #1615), and I hope it can be fixed sometime soon.

Thanks for all of your help, and I hope that if anyone else finds him or herself in a similar situation, they can find a solution here.

Bye! :)
